src/web/transactions.rs

```diff
@@ -23,62 +25,45 @@ pub enum TransactionDTO {
     Rejected(RejectedTransactionDTO),
 }
 
-impl TryFrom<TransactionValue> for TransactionDTO {
+impl TryFrom<TransactionQueryResult> for TransactionDTO {
     type Error = color_eyre::Report;
 
-    fn try_from(tx: TransactionValue) -> Result<Self> {
-        match tx {
-            TransactionValue::Transaction(versioned) => {
-                let tx = versioned.into_v1();
-                let base = TransactionBase::new(tx.hash(), tx.payload, tx.signatures)
-                    .wrap_err("Failed to make TransactionBase")?;
-                Ok(Self::Committed(CommittedTransactionDTO { base }))
-            }
-            TransactionValue::RejectedTransaction(versioned) => {
-                let tx = versioned.into_v1();
-                let hash = tx.hash();
-                let RejectedTransaction {
-                    payload,
-                    signatures,
-                    rejection_reason,
-                } = tx;
-                let base = TransactionBase::new(hash, payload, signatures)
-                    .wrap_err("Failed to make TransactionBase")?;
-                Ok(Self::Rejected(RejectedTransactionDTO {
-                    base,
-                    rejection_reason: rejection_reason.into(),
-                }))
-            }
+    fn try_from(tx_result: TransactionQueryResult) -> Result<Self> {
+        let tx_value = tx_result.transaction();
+        let TransactionValue { tx, error } = tx_value;
+        let hash = tx.hash();
+        let base = TransactionBase::new(hash, tx.payload().clone(), tx.signatures().clone())
```

Comment:
Are you sure that cloning is necessary here?

```suggestion
        let base = TransactionBase::new(hash, tx.payload(), tx.signatures())
```

Reply:
Simply removing ```.clone()``` won't suffice in this scenario because ```tx.payload()``` returns a ```&TransactionPayload```, but the ```TransactionBase``` expects a ```TransactionPayload```. If we want to accept a ```&TransactionPayload```, we'd need to introduce a lifetime parameter in the struct.

However, we can improve the situation by avoiding the explicit need to clone the ```payload``` and ```signatures``` when creating the ```TransactionBase```. Instead, we can modify the function to accept ```references```. Then, during the construction of ```TransactionBase```, we can clone the data from those references.

Reply:
I see. Yup, if `tx.payload()` returns `&TransactionPayload`, than it totally makes sense to simply `.clone()` the value in order to own it in `TransactionBase`.

I am no sure I understand WDYM by "references". Do you mean to change `TransactionBase` structure so that it doesn't own data (like payload), but only keeps a reference to it?

Like this:

```rust
// before
struct TransactionBase {
  payload: TransactionPayload
}

// after
struct TransactionBase<'a> {
  payload: &'a TransactionPayload
}
```